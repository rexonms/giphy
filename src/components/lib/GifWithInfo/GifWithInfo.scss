/** Loading animation like FB and Yelp */
.container {
  position: relative;
  animation-duration: 4s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeHolderShimmer;
  animation-timing-function: linear;
  background: $greyE;
  background: linear-gradient(to right, $greyD 8%, $greyC 18%, $greyD 33%);
  background-size: 800px 104px;
  line-height: 0;
}
@keyframes placeHolderShimmer{
  0%{
    background-position: -100vw 0
  }
  100%{
    background-position: 100vw 0
  }
}
.img  {
  line-height: 0;
  img {
    min-width: 100px;
    min-height: 100px;
  }
}
.imgDetails {
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
  width: 100%;
  background: rgba(0,0,0, .8);
  max-height: 0;
  overflow: hidden;
  transition: max-height 1s;
  div {
    padding: 5px;
  }
  p {
    font-size: $fontSizeSmall;
    color: $white;
    margin: 0px;
    padding: 0px;
  }
}
.show {
  transition: max-height 1s;
  max-height: inherit;
}
