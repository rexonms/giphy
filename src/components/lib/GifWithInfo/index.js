import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Styles from './GifWithInfo.scss';

const propTypes = {
  image: PropTypes.shape({
    url: PropTypes.shape({
      url: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
  rating: PropTypes.string.isRequired,
};

const defaultProps = {};

class GifWithInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDetails: false,
    };
    this.getClassName = this.getClassName.bind(this);
  }
  getClassName() {
    return this.state.showDetails === true ? `${Styles.imgDetails} ${Styles.show}` : Styles.imgDetails;
  }

  render() {
    const { image } = this.props;
    return (
      <div
        className={Styles.container}
        onMouseEnter={() => this.setState({ showDetails: true })}
        onMouseLeave={() => this.setState({ showDetails: false })}
      >
        <div className={Styles.img}>
          <img
            src={image.url.url}
            alt={this.props.title}
            // style={{ height: `${image.url.height}px` }}
          />
        </div>
        <div className={this.getClassName()}>
          <div>
            <p>{this.props.title}</p>
            <p>{this.props.rating}</p>
          </div>
        </div>
      </div>
    );
  }
}

GifWithInfo.propTypes = propTypes;
GifWithInfo.defaultProps = defaultProps;

export default GifWithInfo;
