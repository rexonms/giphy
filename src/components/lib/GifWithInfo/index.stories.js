import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import GifWithInfo from './index';

const stories = storiesOf('lib/GifWithInfo', module);
export const props = {
  id: 'b1YFCoOD2aX2E',
  title: 'frances mcdormand smiling GIF by Maudit',
  rating: 'g',
  user: {
    avatarUrl: 'https://media0.giphy.com/avatars/maudit.gif',
  },
  image: {
    url: {
      url: 'https://media2.giphy.com/media/b1YFCoOD2aX2E/200_d.gif',
      width: '345',
      height: '200',
      size: '353795',
      webp: 'https://media2.giphy.com/media/b1YFCoOD2aX2E/200_d.webp',
      webp_size: '102774',
    },
  },
};

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('GifWithInfo', withInfo('A Stateful Component example that toggles the bulb state based on user click event.')(() => <GifWithInfo {...props} />));
