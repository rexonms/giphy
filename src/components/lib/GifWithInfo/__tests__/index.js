import React from 'react';
// import uuidV4 from 'uuid/v4'
import { shallow } from 'enzyme';

import GifWithInfo from './../index';

describe('GifWithInfo Component', () => {
  let props;
  let shallowVideoComponent;
  const gifWithInfoComponent = () => {
    if (!shallowVideoComponent) {
      shallowVideoComponent = shallow(<GifWithInfo {...props} />);
    }
    return shallowVideoComponent;
  };
  beforeEach(() => {
    props = {
      id: 'b1YFCoOD2aX2E',
      title: 'frances mcdormand smiling GIF by Maudit',
      rating: 'g',
      user: {
        avatarUrl: 'https://media0.giphy.com/avatars/maudit.gif',
      },
      image: {
        url: {
          url: 'https://media2.giphy.com/media/b1YFCoOD2aX2E/200_d.gif',
          width: '345',
          height: '200',
          size: '353795',
          webp: 'https://media2.giphy.com/media/b1YFCoOD2aX2E/200_d.webp',
          webp_size: '102774',
        },
      },
    };
    shallowVideoComponent = undefined;
  });
  it('should render previous snapshot', () => {
    expect(gifWithInfoComponent()).toMatchSnapshot();
  });
  it('should not have .show class on load', () => {
    expect(gifWithInfoComponent().find('.show').exists()).toBe(false);
  });
});

