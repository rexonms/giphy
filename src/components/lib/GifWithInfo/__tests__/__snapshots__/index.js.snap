// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifWithInfo Component should render previous snapshot 1`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="img"
  >
    <img
      alt="frances mcdormand smiling GIF by Maudit"
      src="https://media2.giphy.com/media/b1YFCoOD2aX2E/200_d.gif"
    />
  </div>
  <div
    className="imgDetails"
  >
    <div>
      <p>
        frances mcdormand smiling GIF by Maudit
      </p>
      <p>
        g
      </p>
    </div>
  </div>
</div>
`;
